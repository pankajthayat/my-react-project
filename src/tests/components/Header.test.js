import React from "react"
import ReactShallowRenderer from "react-test-renderer/shallow";
import Header from "../../components/Header";

test("should render Header correctly",()=>{
    const renderer =new ReactShallowRenderer();
    renderer.render(<Header />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();// the 1st time we run this test case it is always going to pass.. cause thr is no existing sanpsot.. so jest will create a new one ...ie... the 1st time header renders. and 2nd time it will try to math that will 1st one
    


    console.log(renderer.getRenderOutput());
})