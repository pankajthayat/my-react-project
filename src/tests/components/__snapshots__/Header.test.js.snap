// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="active-class"
    aria-current="page"
    exact={true}
    to="/"
  >
    Go Home
  </NavLink>
  <NavLink
    activeClassName="active-class"
    aria-current="page"
    to="/create"
  >
     create
  </NavLink>
  <NavLink
    activeClassName="active-class"
    aria-current="page"
    to="/edit"
  >
    Edit
  </NavLink>
  <NavLink
    activeClassName="active-class"
    aria-current="page"
    to="/help"
  >
    help
  </NavLink>
</header>
`;
